go/types/validating_batching_sink.go

```diff
@@ -16,28 +19,65 @@ type ValidatingBatchingSink struct {
 	cs    chunks.ChunkStore
 	batch [batchSize]chunks.Chunk
 	count int
-	tc    *TypeCache
+	pool  sync.Pool
 }
 
-func NewValidatingBatchingSink(cs chunks.ChunkStore, tc *TypeCache) *ValidatingBatchingSink {
-	return &ValidatingBatchingSink{vs: newLocalValueStore(cs), cs: cs, tc: tc}
+func NewValidatingBatchingSink(cs chunks.ChunkStore) *ValidatingBatchingSink {
+	return &ValidatingBatchingSink{
+		vs:   newLocalValueStore(cs),
+		cs:   cs,
+		pool: sync.Pool{New: func() interface{} { return NewTypeCache() }},
```

Comment:
i think the problem with using a pool is that each of these typcaches is going to grow over time (as they accumulate more & more different types).

was it a perf problem to just create a new one for each validating batching sink?